### ISSUE TITLE
.tldr_cache prevents any meaningful output

### ISSUE THREAD

**Issue by ajscop** [tldr-pages/tldr-python-client#31]
Hi, I just installed tldr via the python client on Xubuntu 16. Apart from serious issues with colours that make the text unreadable on every colour scheme, which I have also seen an issue regarding, I have found that the .tldr_cache directory is causing issues for reading tldr pages.

I am not great at explaining in words, so I will just show a few screenshots.

Let's say I run the command `$ tldr nano`. Yes, it comes up with the tldr page for nano. 

However, now if I were to run the same command a second, third, fourth or even 100th time, I will not get any output anymore. I just get three lines of my console coloured with no text.

![image](https://cloud.githubusercontent.com/assets/14885638/18826343/d97939e8-83c4-11e6-8674-5a4782630b2a.png)

This is only for tldr pages that I have previously read. If I run tldr with a different page I have not yet read, it is fine.

The fix seems to be to run `rm -R ~/.tldr_cache`. Once I do this, everything is fine. However, it is annoying to have to keep doing this every single time I want to view a tldr page again. 

![image](https://cloud.githubusercontent.com/assets/14885638/18826501/96f12d32-83c5-11e6-8f8d-be750d3b37fa.png)

Please excuse the text being unreadable. None of the console colour schemes included with the Xubuntu terminal are any better, and I have other stuff which works well with these console colours, so I wont change my console theme just to get tldr to be readable.

(Sorry if this is a bit vague, this is my first issue on another project).

Thanks

**Comment by SpriteMidr** [comment]
@waldyrious glad it is useful. I have not done a report before so I didn't want to give the wrong information. I cant tell if `tldr` is outputting blank lines or text the same colour as the background. I have not got my laptop charged now to find out I am afraid.

**Comment by bwh1te** [comment]
Hello, @SpriteMidr! Please make new screenshot of these commands:
`rm -R .tldr_cache  # if exists
tldr python
cat .tldr_cache`
or just post .tldr_cache content here.

And what to colors... Have you tried to change colors with environment variables?
![20161001_001](https://cloud.githubusercontent.com/assets/946520/19014098/8c0fd1b0-87f4-11e6-8802-54a90302fbf3.png)

Yes, maybe we should change default color scheme. I think it's terrible too :)

**Comment by SpriteMidr** [comment]
Sorry, I had Ubuntu issues and had to reinstall.

**Comment by bwh1te** [comment]
Oh, I'm sorry, was inattentive... Of cource nano instead of python:
```bash
rm -R ~/.tldr_cache  # if exists
tldr nano
ls -la ~/.tldr_cache | grep nano
cat ~/.tldr_cache/nano_common.md
```

**Comment by SpriteMidr** [comment]
![image](https://cloud.githubusercontent.com/assets/14885638/19016843/d270d1e4-881d-11e6-8ec0-3ce22182c1c5.png)

I am going to go to bed now, so I won't be able to reply until tomorrow at some point.

**Comment by bwh1te** [comment]
Interesting. File was successfully created, but was not filled by data. Cannot reproduce this behaviour, but still trying. Judging from the colors, you have the old version of tldr. Could you please update it? If it doesn't help, please try set environment variable `TLDR_CACHE_ENABLED=1`.

I'll think about some refactoring to make tldr a bit more debuggable...

**Comment by SpriteMidr** [comment]
Is the newest update not up on the `python3-pip` repositories? When I ran the most recent screenshot's commands, I had literally just installed `tldr` from the pip3 repositores. This was on an entirely new and clean install of Ubuntu 16.04 LTS. I had switched from xubuntu to Ubuntu MATE. I have had the exact same issues on every PC I have tried this on, which is strange. I have not set any new environment variables or anything since install. It seems a bit strange at how intermittent this issue is :)

Thanks for getting back to me! I appreciate it.

**Comment by NIA** [comment]
@bwh1te  Bump?  Exactly the same here on clean Ubuntu 16.04.1 LTS. Also installed from pip3 (simply no python 2 preinstalled on this version of ubunty, only python 3). Same empty file in .tldr_cache, and same three empty blue lines as on SpriteMidr's screenshots.

**Comment by waldyrious** [comment]
@SpriteMidr, @bwh1te and @NIA -- please confirm than the latest version on master, after merging @iliakonnov's #36, fixes this issue.

**Comment by NIA** [comment]
@waldyrious , I did `pip3 install git+https://github.com/iliakonnov/tldr-python-client.git` right now (as you wrote in comments to #36) on my machine and everything is OK!
![image](https://cloud.githubusercontent.com/assets/26939/23511703/19e23bb2-ff6f-11e6-898b-f13359be830e.png)
Repeated calls work fine, file in cache dir is not empty anymore. Cool 👍 Thanks

**Comment by waldyrious** [comment]
Yeah, we need to make a new release to include this and other fixes. @felixonmars can you do that? (And maybe consider adding [some more people](https://github.com/tldr-pages/tldr-python-client/graphs/contributors) as maintainers on PyPI, if your availability is limited.)

**Comment by SpriteMidr** [comment]
Did the PIP3 repos ever get updated for this fix?
![screenshot from 2017-06-14 08-56-56](https://user-images.githubusercontent.com/14885638/27121484-7cfed018-50df-11e7-99ee-454538278a74.png)

Pulling tldr from python3-pip is just still producing this issue, which is weird as it seems to work fine on my laptop...

**Comment by SpriteMidr** [comment]
...nevermind, I just ran 

```bash
sudo pip3 install git+https://github.com/iliakonnov/tldr-python-client.git
```

...and got my answer. It is working now :)

**Comment by juliohm** [comment]
Any reason why this issue is still not solved in a clean pip install? The solution was already provided in a fork, and yet *every single user* of the package is affected on Python 3.

**Comment by waldyrious** [comment]
@juliohm that's unfortunately blocked on #47. I'm reopening this issue until a new release is made with the fix, to make the situation clearer to end users, since it's a pretty common problem.

**Comment by felixonmars** [comment]
Fixed in 0.4.1.
